bridge/talk/talk.go

```diff
@@ -0,0 +1,116 @@
+package talk
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/42wim/matterbridge/bridge"
+	"github.com/42wim/matterbridge/bridge/config"
+
+	talk "github.com/gary-kim/go-nc-talk"
+	"github.com/gary-kim/go-nc-talk/ocs"
+	"github.com/gary-kim/go-nc-talk/room"
+	"github.com/gary-kim/go-nc-talk/user"
+)
+
+type Btalk struct {
+	user  *user.TalkUser
+	rooms []Broom
+	*bridge.Config
+}
+
+func New(cfg *bridge.Config) bridge.Bridger {
+	return &Btalk{Config: cfg}
+}
+
+type Broom struct {
+	room      *room.TalkRoom
+	ctx       context.Context
+	ctxCancel context.CancelFunc
+}
+
+func (b *Btalk) Connect() (err error) {
+	b.Log.Info("Connecting")
+	b.user = talk.NewUser(b.GetString("Server"), b.GetString("Login"), b.GetString("Password"))
+	_, err = b.user.Capabilities()
+	if err != nil {
+		b.Log.Info("Connected")
+	} else {
+		b.Log.Info("Cannot connect")
+	}
+	return
+}
+
+func (b *Btalk) Disconnect() (err error) {
+	for _, r := range b.rooms {
+		r.ctxCancel()
+	}
+	return
+}
+
+func (b *Btalk) JoinChannel(channel config.ChannelInfo) (err error) {
+	newRoom := Broom{
+		room: talk.NewRoom(b.user, channel.Name),
+	}
+	newRoom.ctx, newRoom.ctxCancel = context.WithCancel(context.Background())
+	c, err := newRoom.room.ReceiveMessages(newRoom.ctx)
+	if err != nil {
+		return
+	}
+	b.rooms = append(b.rooms, newRoom)
+	go func() {
+		for msg := range c {
+			// ignore messages that are one of the following
+			// * not a message from a user
+			// * from ourselves
+			if msg.MessageType != ocs.MessageComment || msg.ActorID == b.user.User {
+				continue
+			}
+			remoteMessage := config.Message{
+				Text:      msg.Message,
+				Channel:   newRoom.room.Token,
+				Username:  msg.ActorDisplayName,
+				UserID:    msg.ActorID,
+				Account:   b.Account,
+				Timestamp: time.Unix(int64(msg.Timestamp), 0),
```

Comment:
You don't need to set the Timestamp here, the gateway does this

Reply:
Okay, thank you!